src/libraries/System.Private.CoreLib/src/System/Guid.cs

```diff
@@ -844,43 +852,39 @@ private static ReadOnlySpan<char> EatAllWhitespace(ReadOnlySpan<char> str)
             (str[i + 1] | 0x20) == 'x';
 
         // Returns an unsigned byte array containing the GUID.
-        public byte[] ToByteArray()
+        public byte[] ToByteArray() => ToByteArray(false);
+
+        // Returns an unsigned byte array containing the GUID.
+        public byte[] ToByteArray(bool bigEndian)
         {
             var g = new byte[16];
-            if (BitConverter.IsLittleEndian)
-            {
-                MemoryMarshal.TryWrite(g, ref Unsafe.AsRef(in this));
-            }
-            else
-            {
-                TryWriteBytes(g);
-            }
+            TryWriteBytes(g, bigEndian, out _);
             return g;
         }
 
         // Returns whether bytes are successfully written to given span.
-        public bool TryWriteBytes(Span<byte> destination)
+        public bool TryWriteBytes(Span<byte> destination) => TryWriteBytes(destination, false, out _);
```

Comment:
This has the same perf problem as constructor used to have. For example, `static bool Test(Span<byte> span) => return new Guid().TryWriteBytes(span);` is 4x slower in microbenchmark.